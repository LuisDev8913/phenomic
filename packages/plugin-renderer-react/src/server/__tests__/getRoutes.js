// @flow

import * as React from "react";

import getRoutes from "../getRoutes";

const Noop = () => {
  return null;
};
const noop = () => {};
it("should resolve URLs based on routes", async () => {
  const Router = Noop;
  const Route = Noop;
  const browserHistory = noop;
  const Wrapper = Noop;
  const Home = Noop;
  const DocPage = Noop;
  const PageError = Noop;

  expect(
    await getRoutes({
      routes: (
        <Router history={browserHistory}>
          <Route component={Wrapper}>
            <Route path="/" component={Home} />
            <Route path="/docs/*" component={DocPage} />
            <Route path="*" component={PageError} />
          </Route>
        </Router>
      ),
    }),
  ).toMatchSnapshot();
});
