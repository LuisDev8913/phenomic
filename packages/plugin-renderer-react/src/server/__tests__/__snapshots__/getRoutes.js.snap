// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should resolve URLs based on routes 1`] = `
Array [
  Object {
    "component": [Function],
    "path": "/",
  },
  Object {
    "component": [Function],
    "path": "/docs/*",
  },
  Object {
    "component": [Function],
    "path": "/*",
  },
]
`;
